src/sdk/namespace_s3.js

```diff
@@ -745,6 +745,43 @@ class NamespaceS3 {
         throw new S3Error(S3Error.NotImplemented);
     }
 
+    //////////////////////////
+    //  OBJECT ATTRIBUTES   //
+    //////////////////////////
+
+    async get_object_attributes(params, object_sdk) {
+        dbg.log0('NamespaceS3.get_object_attributes:', this.bucket, inspect(params));
+        await this._prepare_sts_client();
+
+        /** @type {AWS.S3.GetObjectAttributesRequest} */
+        const request = {
+            Bucket: this.bucket,
+            Key: params.key,
+            VersionId: params.version_id,
+            ObjectAttributes: [params.attributes.join(',')],
```

Comment:
Check if this is the right parameter, I expect it to be an array of strings where each item is a single attribute name.

Reply:
I tried to change one of my tests, and I saw that I receive `Invalid Argument` from the SDK.
The example:
```
mocha.it("get object attributes - check version-id header", async function() {
            s3_uid6.middlewareStack.add(
                next => async args => {
                  const result = await next(args);
                  result.output.$metadata.headers = result.response.headers;
                  return result;
                }
              );

            const res = await s3_uid6.getObjectAttributes({
                Bucket: bucket_name,
                Key: key,
                VersionId: version_id,
                ObjectAttributes: ['ETag', 'ObjectSize', 'StorageClass'],
            });
            assert.ok(res.$metadata.httpStatusCode === 200);
            assert.equal(res.$metadata.headers['x-amz-version-id'], version_id);
        });
```
the line that was changed:
From: ObjectAttributes: ['ETag,ObjectSize,StorageClass'], (array with a single string with the attributes)
To: ObjectAttributes: ['ETag', 'ObjectSize', 'StorageClass'], (array of strings)

The error:
```
1) bucketspace namespace_fs - versioning
       get object attributes
         get object attributes - check version-id header:
     InvalidArgument: Invalid Argument
      at throwDefaultError (node_modules/@smithy/smithy-client/dist-cjs/index.js:842:20)
      at /Users/shiradymnik/SourceCode/noobaa-core/node_modules/@smithy/smithy-client/dist-cjs/index.js:851:5
      at de_CommandError (node_modules/@aws-sdk/client-s3/dist-cjs/index.js:4751:14)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async /Users/shiradymnik/SourceCode/noobaa-core/node_modules/@smithy/middleware-serde/dist-cjs/index.js:35:20
      at async /Users/shiradymnik/SourceCode/noobaa-core/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:482:18
      at async /Users/shiradymnik/SourceCode/noobaa-core/node_modules/@smithy/middleware-retry/dist-cjs/index.js:320:38
      at async /Users/shiradymnik/SourceCode/noobaa-core/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:110:22
      at async /Users/shiradymnik/SourceCode/noobaa-core/src/test/unit_tests/test_bucketspace_versioning.js:1043:34
      at async /Users/shiradymnik/SourceCode/noobaa-core/src/test/unit_tests/test_bucketspace_versioning.js:929:34
      at async /Users/shiradymnik/SourceCode/noobaa-core/src/test/unit_tests/test_bucketspace_versioning.js:914:34
      at async /Users/shiradymnik/SourceCode/noobaa-core/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:138:14
      at async /Users/shiradymnik/SourceCode/noobaa-core/node_modules/@aws-sdk/middleware-logger/dist-cjs/index.js:34:22
      at async Context.<anonymous> (src/test/unit_tests/test_bucketspace_versioning.js:1049:25)
```

At this point, I will not change it.

Reply:
But it complains about the version id header ("check version-id header")... not about the attributes...

Reply:
It is the name of the test

Reply:
I just tested it, and I see the exact opposite behavior, it works for me only with: 
```
ObjectAttributes: ["ETag", "ObjectSize", "StorageClass"]
```

But fails with:
```
ObjectAttributes: ["ETag,ObjectSize,StorageClass"]
```

with the following error:
```
./node_modules/@smithy/smithy-client/dist-cjs/index.js:846
  const response = new exceptionCtor({
                   ^

InvalidArgument: Invalid attribute name specified.
    at throwDefaultError (./node_modules/@smithy/smithy-client/dist-cjs/index.js:846:20)
    at ./node_modules/@smithy/smithy-client/dist-cjs/index.js:855:5
    at de_CommandError (./node_modules/@aws-sdk/client-s3/dist-cjs/index.js:4764:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async ./node_modules/@smithy/middleware-serde/dist-cjs/index.js:35:20
    at async ./node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:484:18
    at async ./node_modules/@smithy/middleware-retry/dist-cjs/index.js:320:38
    at async ./node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:110:22
    at async ./node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:138:14
    at async ./node_modules/@aws-sdk/middleware-logger/dist-cjs/index.js:34:22 {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 400,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  Code: 'InvalidArgument',
  ArgumentName: 'x-amz-object-attributes',
  ArgumentValue: '"ETag',
}
```

Reply:
I will check, thanks.

Reply:
I changed it to an array of strings.
Note that the `Invalid Argument` was due to parsing that was missing `trim()` for AWD SDK client (AWS CLI client didn't need it).

Reply:
But why do I still see you are using join(',') here...?

Reply:
It is an outdated part, in the updated code I don't use it anymore.